### PR TITLE
Convert `cluster.routing.allocation.type` and `processors` to the new settings infra.

### PR DESCRIPTION
**Pull request by s1monw**
this also removes some unused / unsupported settings from the tests.

### CODE REVIEW

**Comment by jpountz** [comment]
LGTM

File: test/framework/src/main/java/org/elasticsearch/test/InternalTestCluster.java [hunk 1]
```diff
@@ -392,11 +392,7 @@ private Settings getRandomNodeSettings(long seed) {
             builder.put(SearchService.DEFAULT_KEEPALIVE_SETTING.getKey(), TimeValue.timeValueSeconds(100 + random.nextInt(5 * 60)));
         }
 
-        if (random.nextInt(10) == 0) {
-            // node gets an extra cpu this time
-            builder.put(EsExecutors.PROCESSORS, 1 + EsExecutors.boundedNumberOfProcessors(Settings.EMPTY));
-        }
-
+        builder.put(EsExecutors.PROCESSORS_SETTING.getKey(), 1 + random.nextInt(3));
```
Comment (javanna) [inline test/framework/src/main/java/org/elasticsearch/test/InternalTestCluster.java]: is the effect the same as before? we were changing this only rarely before while now we do it all the time?

**Comment by javanna** [comment]
left a small comment LGTM otherwise

File: test/framework/src/main/java/org/elasticsearch/test/InternalTestCluster.java (hunk 1 above)
Comment (s1monw) [inline test/framework/src/main/java/org/elasticsearch/test/InternalTestCluster.java]: it was set all the time via system property to make tests reproducible. We are now doing this too.

Comment (javanna) [inline test/framework/src/main/java/org/elasticsearch/test/InternalTestCluster.java]: cool ok.

**Pull request merged**